File: app/src/main/java/com/galib/natorepbs2/db/ComplainCentreDao.java
package com.galib.natorepbs2.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;
@Dao
public interface ComplainCentreDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(ComplainCentre complainCentre);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<ComplainCentre> complainCentreList);

    @Query("SELECT * FROM complain_center_table ORDER BY priority ASC")
    LiveData<List<ComplainCentre>> getAll();

    @Query("DELETE FROM complain_center_table")
    void deleteAll();
}

File: app/src/main/java/com/galib/natorepbs2/db/ComplainCentreDao.kt
package com.galib.natorepbs2.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.galib.natorepbs2.models.ComplainCentre
import kotlinx.coroutines.flow.Flow

@Dao
interface ComplainCentreDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(complainCentre: ComplainCentre)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(complainCentreList: List<ComplainCentre>)

    @get:Query("SELECT * FROM complain_center_table ORDER BY priority ASC")
    val allComplainCentre: Flow<List<ComplainCentre>>

    @Query("DELETE FROM complain_center_table")
    suspend fun deleteAll()
}